avidemux_plugins/ADM_videoEncoder/x265/qt4/Q_x265.cpp

```diff
@@ -122,6 +122,60 @@ bool x265_ui(x265_settings *settings)
 
     return success;
 }
+
+/**
+    \brief Set combo box items to be elements in container with data.
+
+    Elements of container should be idcToken.  The string will be the item name
+    and value will be added as item data.
+
+    One could design function that used a lambda to convert from element type
+    into label and value, so that any element type can be used, but that is
+    C++14.
+
+    If initial is non-NULL, it will be added first and the data used will be
+    initdata;
+
+    As a somewhat ugly hack for Qt4, the initial string will go through
+    fromUtf8() in case it has been translated to non-ASCII.  As none of the
+    items in the containers used are translated, they don't need this.
+
+    \param [in] box ComboBox to set items in
+    \param [in] items Container (array) of idcToken to use.
+    \param [in] initial An option first item to add.
+    \param [in] initdata The data to be assigned with optional first item, default is empty.
+*/
+template <typename ContainerT>
+static void fillComboBoxData(QComboBox *box, const ContainerT& items, const char* initial = NULL, const QVariant& initdata = QVariant())
+{
+    box->clear();
+    if (initial)
+        box->addItem(QString::fromUtf8(initial), initdata);
+    for (const idcToken *t = std::begin(items); t != std::end(items); t++)
```

Comment:
In C++11, I would write `const auto iter = std::begin(items)` and then it wouldn't matter kind of container or exactly what type was in the container.   One could call it with an array of {std::string, int} and again with a std::vector of {QString, unsigned long}, and it would work with both.  But making that work in C++98 is just too much of a pain.

Reply:
As far as I can tell, there is no problem with using C++11 features, it is just necessary to explicitly set `-std=c++11` else Apple clang doesn't play nice.